File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n=nums.size();
        if(nums.empty()) return 0;
        sort(nums.begin(),nums.end());
        int x=nums[0];
        int cnt=1;
        int maxlen=1;
        for(int i=1;i<n;i++)
        {
            if(nums[i]==x+1){
                x+=1;
                cnt++;
                maxlen=max(maxlen,cnt);
            }
            else if(nums[i]!=x){
                x=nums[i];
                cnt=1;
            }
        }
        return maxlen;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {

        HashSet<Integer> set = new HashSet<>();

        // Put all elements into HashSet
        for (int x : nums) {
            set.add(x);
        }

        int maxLen = 0;

        for (int x : set) {

            // x is the starting point of a sequence
            if (!set.contains(x - 1)) {

                int curr = x;
                int count = 1;

                while (set.contains(curr + 1)) {
                    curr++;
                    count++;
                }

                maxLen = Math.max(maxLen, count);
            }
        }

        return maxLen;
    }
}